### DISCUSSION TITLE
Setting `$GITHUB_PATH` through `bash` doesn't carry over to `powershell`

### DISCUSSION THREAD

**Discussion by ok-nick** [community/community#29983] [Actions]
Hi, I recently received a [pull request on one of my repos](https://github.com/ok-nick/setup-aftman/pull/1) where an environment variable set through `bash` does not replicate to `powershell`.  

[Here](https://github.com/ok-nick/setup-aftman/blob/14933ea774fbc37dfef0bf7033b24034d1159069/action.yml#L53) I set the environment variable via `echo "$HOME/.aftman/bin" >> $GITHUB_PATH` in `bash`. If I were to use `powershell` in the next step to try an access any executables under that directory it would not work. However, if I were to use `bash` it would work fine.

Am I missing something? Is this normal behavior? Is setting the environment variable via `powershell` the only way to solve this?

**Comment by aayushchugh** [comment]
Yes this is normal behavior. because you are setting environment variables in Bash and trying to access them in PowerShell these both are two different things
these are [Docs](https://shellgeek.com/set-environment-variable-using-powershell/#:~:text=Set%20Environment%20Variable%20using%20PowerShell%201%20Set%20Environment,value%20to%20Environment%20Variable%20...%205%20Conclusion%20) how you can add env variable in powershell

**Reply by airtower-luna** [reply]
Each step gets a fresh shell anyway, and the Actions runner transfers things written to the environment files between them, so that shouldn't matter. I've used environment variables between different shells (but I don't use Powershell).

**Comment by airtower-luna** [comment]
Just a guess: I wonder if the different path formats used by Bash and Powershell might mess things up? I know Bash uses a Unix-ish path format even when running on Windows, maybe Powershell can't understand the result?

**Reply by ok-nick** [reply]
Perhaps. Is there a suggested method for making a path Windows compatible in bash?

**Reply by airtower-luna** [reply]
Not that I'm aware of, but then again I don't use Windows myself. :sweat_smile: 

My first try would be some simple search and replace, but there might be some pitfalls (e.g. this would also replace escaped slashes):
```
echo $YOUR_PATH | sed -e 's,/\([a-z]\)/,\1:\\,' -e 's,/,\\,g'
```

**Reply by ok-nick** [reply]
I'll have to try it out, thanks.
